pkg/util/utils_supported.go

```diff
@@ -60,6 +60,36 @@ func GetRootlessRuntimeDir() (string, error) {
 	return rootlessRuntimeDir, nil
 }
 
+// GetRootlessConfigHomeDir returns the config home directory when running as non root
```

Comment:
I'd simplify this method to the equivalent of, or re-use [homedir.GetConfigHome](https://github.com/moby/moby/blob/master/pkg/homedir/homedir_linux.go#L100) (see other comments on this function).

The use of [homedir.Get](https://github.com/moby/moby/blob/master/pkg/homedir/homedir_unix.go#L20) elsewhere also suggests that this could be simplified further to get rid of the error case if we fail early when`$HOME` is not defined &mdash; other code paths might already depend on it anyway.

Reply:
I looked at that but it doesn't create the directory if it's not there and I think we need that.